src/service/impl/departures/utils/favorites.ts

```diff
@@ -79,3 +82,47 @@ export function filterStopPlaceFavorites(
     },
   };
 }
+
+function favoriteDepartureMatchesEstimatedCall(
+  favoriteDeparture: FavoriteDeparture,
+  estimatedCall: EstimatedCall,
+) {
+  return (
+    estimatedCall.serviceJourney?.line.id === favoriteDeparture.lineId &&
+    (!favoriteDeparture.destinationDisplay ||
+      destinationDisplaysAreMatching(
+        favoriteDeparture.destinationDisplay,
+        estimatedCall.destinationDisplay,
+      )) &&
+    (!favoriteDeparture.lineName ||
+      favoriteDeparture.lineName ===
+        mapToLegacyLineName(estimatedCall.destinationDisplay))
+  );
+}
+
+/* NB this is the same function as in the app. Keep in sync! */
```

Comment:
Will this be an issue if the app changes something in a future version, and it becomes out of sync with the BFF?

Reply:
"Keep in sync" was maybe not the best comment here.
`destinationDisplaysAreEqual` is just a simple util for checking if `destinationDisplay1 === destinationDisplay2`, without any "smart" behaviour.
So it would not be of great concern

Reply:
What's important is what we send in to this function